digest_test.go

```diff
@@ -37,41 +37,47 @@ func TestH(t *testing.T) {
 			"r2":          "GET:/dir/index.html",
 			"expected_r1": "939e7578ed9e3c518a452acee763bce9",
 			"expected_r2": "39aff3a2bab6126f332b942af96d3366",
-			"expected_r3": "6629fae49393a05397450978507c4ef1",
+			"expected_r3": "72e6fcaeee7554ad095e1147cd8f3f13",
 		},
 		"SHA-256": {
 			"r1":          "Mufasa:testrealm@host.com:Circle Of Life",
 			"r2":          "GET:/dir/index.html",
 			"expected_r1": "3ba6cd94661c5ef34598040c868f13b8775df29109986be50ad35ae537dd3aa4",
 			"expected_r2": "9a3fdae9a622fe8de177c24fa9c070f2b181ec85e15dcbdc32e10c82ad450b04",
-			"expected_r3": "5abdd07184ba512a22c53f41470e5eea7dcaa3a93a59b630c13dfe0a5dc6e38b",
+			"expected_r3": "705e2bcd4503344546454cc767a5e3107dc09f86dc898d840fff18b257f0b62f",
 		},
 	}
 
 	for testType, values := range testCases {
-		t.Run(testType, func(t *testing.T) {
+		r1 := values["r1"]
```

Comment:
I enabled a linter for this, loops + function in go can tricky with pointers as they';; keep the reference to the last element of the loop, this also exposed a fail in the tests for the same reason